src/box/lua/space.cc

```diff
@@ -271,6 +272,23 @@ lbox_fillspace(struct lua_State *L, struct space *space, int i)
 	lua_pushcfunction(L, lbox_space_before_replace);
 	lua_settable(L, i);
 
+	lua_pushstring(L, "upgrade_state");
+	if (space->def->opts.upgrade != NULL) {
+		lua_newtable(L);
+
+		struct space_upgrade *upgrade = space->def->opts.upgrade;
+		lua_pushstring(L, upgrade_status_strs[upgrade->status]);
+		lua_setfield(L, -2, "status");
+
+		if (upgrade->func != NULL) {
+			lua_pushstring(L, upgrade->func->def->name);
+			lua_setfield(L, -2, "function");
+		}
+	} else {
+		lua_pushnil(L);
+	}
+	lua_settable(L, i);
```

Comment:
It is done on purpose (if I get you). In case we push new upgrade state only if space->upgrade is not null then it won't get updated after upgrade is finished.